GPU/Common/GPUStateUtils.cpp

```diff
@@ -572,3 +572,462 @@ void ConvertViewportAndScissor(bool useBufferedRendering, float renderWidth, flo
 #endif
 	}
 }
+
+static const BlendFactor genericALookup[11] = {
+	BlendFactor::DST_COLOR,
+	BlendFactor::ONE_MINUS_DST_COLOR,
+	BlendFactor::SRC_ALPHA,
+	BlendFactor::ONE_MINUS_SRC_ALPHA,
+	BlendFactor::DST_ALPHA,
+	BlendFactor::ONE_MINUS_DST_ALPHA,
+	BlendFactor::SRC_ALPHA,			// GE_SRCBLEND_DOUBLESRCALPHA
+	BlendFactor::ONE_MINUS_SRC_ALPHA,		// GE_SRCBLEND_DOUBLEINVSRCALPHA
+	BlendFactor::DST_ALPHA,			// GE_SRCBLEND_DOUBLEDSTALPHA
+	BlendFactor::ONE_MINUS_DST_ALPHA,		// GE_SRCBLEND_DOUBLEINVDSTALPHA
+	BlendFactor::CONSTANT_COLOR,		// FIXA
+};
+
+static const BlendFactor genericBLookup[11] = {
+	BlendFactor::SRC_COLOR,
+	BlendFactor::ONE_MINUS_SRC_COLOR,
+	BlendFactor::SRC_ALPHA,
+	BlendFactor::ONE_MINUS_SRC_ALPHA,
+	BlendFactor::DST_ALPHA,
+	BlendFactor::ONE_MINUS_DST_ALPHA,
+	BlendFactor::SRC_ALPHA,			// GE_SRCBLEND_DOUBLESRCALPHA
+	BlendFactor::ONE_MINUS_SRC_ALPHA,		// GE_SRCBLEND_DOUBLEINVSRCALPHA
+	BlendFactor::DST_ALPHA,			// GE_SRCBLEND_DOUBLEDSTALPHA
+	BlendFactor::ONE_MINUS_DST_ALPHA,		// GE_SRCBLEND_DOUBLEINVDSTALPHA
+	BlendFactor::CONSTANT_COLOR,		// FIXB
+};
+
+static const BlendEq eqLookupNoMinMax[] = {
+	BlendEq::ADD,
+	BlendEq::SUBTRACT,
+	BlendEq::REVERSE_SUBTRACT,
+	BlendEq::ADD,			// GE_BLENDMODE_MIN
+	BlendEq::ADD,			// GE_BLENDMODE_MAX
+	BlendEq::ADD,			// GE_BLENDMODE_ABSDIFF
+};
+
+static const BlendEq eqLookup[] = {
+	BlendEq::ADD,
+	BlendEq::SUBTRACT,
+	BlendEq::REVERSE_SUBTRACT,
+	BlendEq::MIN,			// GE_BLENDMODE_MIN
+	BlendEq::MAX,			// GE_BLENDMODE_MAX
+	BlendEq::ADD,			// GE_BLENDMODE_ABSDIFF
+};
+
+static BlendFactor toDualSource(BlendFactor blendfunc) {
+	switch (blendfunc) {
+#if !defined(USING_GLES2)   // TODO: Remove when we have better headers
+	case BlendFactor::SRC_ALPHA:
+		return BlendFactor::SRC1_ALPHA;
+	case BlendFactor::ONE_MINUS_SRC_ALPHA:
+		return BlendFactor::ONE_MINUS_SRC1_ALPHA;
+#endif
+	default:
+		return blendfunc;
+	}
+}
+
+static BlendFactor blendColor2Func(u32 fix, bool &approx) {
+	if (fix == 0xFFFFFF)
+		return BlendFactor::ONE;
+	if (fix == 0)
+		return BlendFactor::ZERO;
+
+	// Otherwise, it's approximate if we pick ONE/ZERO.
+	approx = true;
+
+	const Vec3f fix3 = Vec3f::FromRGB(fix);
```

Comment:
With the pre-src fixed blending (which I think is completely safe), I think that this check is no longer a good idea.  Maybe we should just remove.

-[Unknown]